File: app/src/main/java/com/boardgamegeek/ui/dialog/CollectionFilterDialog.java
package com.boardgamegeek.ui.dialog;

import android.content.Context;

import com.boardgamegeek.filterer.CollectionFilterer;

public interface CollectionFilterDialog {
	interface OnFilterChangedListener {

		void removeFilter(int type);

		void addFilter(CollectionFilterer filter);
	}

	void createDialog(Context context, OnFilterChangedListener listener, CollectionFilterer filter);

	int getType(Context context);
}

File: app/src/main/java/com/boardgamegeek/ui/dialog/CollectionFilterDialog.kt
package com.boardgamegeek.ui.dialog

import android.content.Context
import androidx.fragment.app.FragmentActivity
import com.boardgamegeek.filterer.CollectionFilterer

interface CollectionFilterDialog {
    fun createDialog(activity: FragmentActivity, filter: CollectionFilterer?)

    fun getType(context: Context): Int
}
